pkg/sql/row/errors.go

```diff
@@ -96,12 +96,16 @@ func NewUniquenessConstraintViolationError(
 			"duplicate key value: decoding err=%s", err)
 	}
 
+	// Exclude implicit partitioning columns from the error message.
+	numImplicitCols := index.Partitioning.NumImplicitColumns
```

Comment:
if we want to exclude hash sharded indexes from displaying as well, we could use `ExplicitColumnStartIdx()`